functions/exponents_and_logs.py

```diff
@@ -1,76 +1,209 @@
+        # Now that y is even, we can use the fact that x^y = (x^2)^(n/2) when y is even
+        base *= base
+        exponent >>= 1
+    return result
+
+
+def radical(radicand: float, index: int, delta: float = 1E-10) -> float:
+    """Returns an approximation of the real value of some radical using Newton's method for finding roots of functions if such a value exists.
+    
+    Value returned is positive if a positive radical exists and negative otherwise. If no real root exists, an exception is raised.
+
+    Args:
+        radicand (float): Radicand of radical to be determined
+        index (int): Index of radical to be determined
+        delta (float): Error tolerance
+
+    Returns:
+        float: Approximation of radical defined by arguments
+        
+    Raises:
+        InputError: If no real root exists for provided arguments
+    """
+    # Ensure that real value of root exists.
+    if common.is_negative(radicand) and common.is_even(index):
+        raise exceptions.InputError(None, "Radical with negative radicand and even index is undefined over set of real numbers.")
+        
+    # Approximate radicand using Newton's method.
+    approx = 1
+    old_mantissa, _ = math.frexp(approx)
+    while True:
+        power = pow_int(approx, index)
+        approx = (index - 1 + radicand / power) * (approx / index)
+        
+        # Return current approximation when the absolute difference is within the desired range.
+        diff = abs(power - radicand)
+        if diff < delta:
+            return approx
+            
+        # Return early if approximation stops improving
+        crrt_mantissa, _ = math.frexp(approx)
+        diff = abs(old_mantissa - crrt_mantissa)
+        # We use 1E-15 here because a double precision floating point mantissa can hold no more than 52 bits of precision.
+        if diff < 1E-15:
+            return approx
+        old_mantissa = crrt_mantissa
+
+def exp_taylor(exponent: float = 1, term_count: int = 50) -> float:
+    """Returns an approximation of the value of e to the power of some real number 'exponent' using the Maclaurin series expansion of e^x.
+    
+    This function is typically only used with values of 'exponent' close to 0 (the center of the Maclaurin series). 
+    For further values, a larger number of terms is required for convergence.
+    
+    Args:
+        exponent (float): Exponent used in power of e being approximated
+        term_count (int): Number of terms used in approximation of power of e
+
+    Returns:
+        float: Approximation of e to the power of 'exponent'
+    """
+    result = 1
     numerator = 1
     denominator = 1
+    for i in range(1, term_count):
+        numerator *= exponent
+        denominator *= i
+        result += numerator / denominator
+        
+    return result
 
-    for n in range(1, 50):
-        numerator *= x
-        denominator *= n
-        e += numerator / denominator
-    return e
 
+def exp(exponent: float = 1) -> float:
```

Comment:
Function should not be named exp if its calculating e^x, this isn't intuitive and can be quite confusing

Reply:
I thought that generate_e was misleading because it can generate any power of e (not necessarily just e itself). I thought of changing it to generate_power_of_e but then I decided to just follow the naming scheme used by math libraries like Python and Java's. They both use exp to refer to powers of e and pow to refer to powers in general. I figured that this was both intuitive and short enough to also be used as an identifier in the calculator interface.

Reply:
I agree the original was misleading but for those unfamiliar with the inner workings of the Python/Java math libs, exp doesn't clearly articulate we're using e. I would prefer to use generate_power_of_e and generate_e for e^x and e respectively. If we decide to create a separate file for pi/e and other constants, having the generate_%constant% convention would be more readable than using a distinct naming strategy for e. 

I don't consider this a major issue though. Lets run it by a few more people. If its nor confusing to the majority of us, I'm fine with leaving it as is. I just don't know how familiar the TAs may be of these conventions.

Reply:
I could get with using generate_power_of_e for the sake of consistency. I had kind of started doing something similar with pow_pi and pow_10. Still though, we would need to choose a name for the calculator interface itself. I like exp because it's short and established by other libraries but if it isn't intuitive to the others then I'm probably alone on this. Maybe we could change it to pow_e or something?

Reply:
I like the prefix `pow` with specifics after

Reply:
Sounds good, I will adopt this convention.